text/0617-cloudfront-oac-l2.md

```diff
@@ -0,0 +1,877 @@
+          throw new Error(`Origin access control for an S3 origin must have origin type
+          '${cloudfront.OriginAccessControlOriginType.S3}', got origin type
+          '${this.originAccessControl.originAccessControlOriginType}'`);
+        }
+
+        const distributionId = options.distributionId;
+        const result = this.grantDistributionAccessToBucket(distributionId);
+
+        // Failed to update bucket policy, assume using imported bucket
+        if (!result.statementAdded) {
+          if (props.overrideImportedBucketPolicy) {
+            this.grantDistributionAccessToImportedBucket(scope, distributionId);
+          } else {
+            Annotations.of(scope).addWarningV2('@aws-cdk/aws-cloudfront-origins:updateBucketPolicy',
+              'Cannot update bucket policy of an imported bucket. Set overrideImportedBucketPolicy to true or update the policy manually instead.');
+          }
+        }
+
+        if (this.bucket.encryptionKey) {
+          this.grantDistributionAccessToKey(scope, distributionId, this.bucket.encryptionKey);
+        }
+
+        const originBindConfig = this._bind(scope, options);
+
+        // Update configuration to set OriginControlAccessId property
+        return {
+          ...originBindConfig,
+          originProperty: {
+            ...originBindConfig.originProperty!,
+            originAccessControlId: this.originAccessControl.originAccessControlId,
+          },
+        };
+      }
+
+      /**
+      * If you're using origin access control (OAC) instead of origin access identity, specify an empty `OriginAccessIdentity` element.
+      * @see http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-s3originconfig.html#cfn-cloudfront-distribution-s3originconfig-originaccessidentity
+      */
+      protected renderS3OriginConfig(): cloudfront.CfnDistribution.S3OriginConfigProperty | undefined {
+        return { originAccessIdentity: '' };
+      }
+
+      private grantDistributionAccessToBucket(distributionId: string): iam.AddToResourcePolicyResult {
+        const oacReadOnlyBucketPolicyStatement = new iam.PolicyStatement(
+          {
+            effect: iam.Effect.ALLOW,
+            principals: [new iam.ServicePrincipal('cloudfront.amazonaws.com')],
+            actions: ['s3:GetObject'],
+            resources: [this.bucket.arnForObjects('*')],
+            conditions: {
+              StringEquals: {
+                'AWS:SourceArn': `arn:${Aws.PARTITION}:cloudfront::${Aws.ACCOUNT_ID}:distribution/${distributionId}`,
+              },
+            },
+          },
+        );
+        const result = this.bucket.addToResourcePolicy(oacReadOnlyBucketPolicyStatement);
+        return result;
+      }
+
+      /**
+       * Use custom resource to update bucket policy
+       */
+      private grantDistributionAccessToImportedBucket(scope: Construct, distributionId: string) {
```

Comment:
Please remove this entirely. CDK does set the expectation that imported resources will be modifiable without CDK migrate. If this isn't made crystal clear by our developer guide, it would be worth updating.